pkg/ipcache/ipcache.go

```diff
@@ -488,9 +526,9 @@ func (ipc *IPCache) UpsertPrefixes(prefixes []netip.Prefix, src source.Source, r
 	ipc.metadata.Lock()
 	for _, p := range prefixes {
 		ipc.metadata.upsertLocked(p, src, resource, labels.GetCIDRLabels(p))
-		ipc.metadata.enqueuePrefixUpdates(p)
 	}
 	ipc.metadata.Unlock()
+	ipc.metadata.enqueuePrefixUpdates(prefixes...)
```

Comment:
I just noticed that there's an lock ordering change here as well. Previously `enqueuePrefixUpdates` was called within the metadata lock, and now it's being done outside. It's worth checking whether this won't have unintended side effects.

Reply:
I stared at the wall for a while and couldn't think of any. The prefix update loop has an entirely independent lifecycle, so any possible interleaving of removal and updates is still possible. In other words, async gonna async.

That is to say, `(removeMetadata(A), enqueue(a), removeMetadata(b), enqueue(b))` can result in `(inject(a), inject(b))` **or** `inject(a, b)` depending on the status of the ipcache injection controller.

I'm pretty sure the original code was unnecessary. All other functions save `enqueuePrefixUpdates()` for the end, including Upsert, which would _theoretically_ be more timing sensitive.

Reply:
I noticed and pondered this earlier too, and came to the same conclusions as Casey. Given that `TriggerLabelInjection()` and the dequeue/enqueue functions for the prefix updates don't interact with the metadata lock, I couldn't identify any meaningful impact to ordering.

Strictly speaking it's plausible for some of these updates to be handled by `InjectLabels()` before the prefix is enqueued, iff the prefix was enqueued previously and `InjectLabels()` is starting to process the updates immediately after the `ipc.metadata.Unlock()` above. The result would be that the next round for running `InjectLabels()` would end up handling this prefix and doing nothing for it. I believe this should be idempotent. If we see an issue there then we could iterate further on the locking pattern or handling logic, but I think we can count that as beyond the scope of this PR.